sensible-proxy.go

```diff
@@ -14,39 +14,37 @@ import (
 	"net"
 	"os"
 	"os/signal"
-	"strconv"
 	"strings"
 	"syscall"
 )
 
+var appLog *log.Logger
+
 func main() {
 	// default configuration
 	var (
-		httpPort  int = 80
-		httpsPort int = 443
+		httpPort   string = "80"
+		httpsPort  string = "443"
+		appLogPath string = "/var/log/sensible-proxy.log"
 	)
 
-	// Remove the default prefixing for log output, rely on process
-	// supervisor to prefix the timestamp and process name
-	log.SetFlags(0)
-
 	// Get configuration from ENV
-	var err error
 	if envVarSet("HTTP_PORT") {
-		httpPort, err = intFromEnvVar("HTTP_PORT")
-		if err != nil {
-			log.Printf("Could not convert ENV variable HTTP_PORT to an integer")
-		}
+		httpPort = os.Getenv("HTTP_PORT")
 	}
 	if envVarSet("HTTPS_PORT") {
-		httpsPort, err = intFromEnvVar("HTTPS_PORT")
-		if err != nil {
-			log.Printf("Could not convert ENV variable HTTPS_PORT to an integer")
-		}
+		httpsPort = os.Getenv("HTTPS_PORT")
+	}
+	if envVarSet("LOG_PATH") {
+		appLogPath = os.Getenv("LOG_PATH")
 	}
+
+	logFile, err := os.OpenFile(appLogPath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
 	if err != nil {
+		log.Fatalln("Failed to open log file", err)
```

Comment:
log.Fatal() will exit the program for you, see https://golang.org/src/log/log.go#L302